backend/src/services/pattern_extraction_service.py

```
@@ -1,134 +1,698 @@
+            return None
+        
+        # Normalize description for matching
+        normalized_desc = description.upper().strip()
+        
+        # Remove common prefixes
+        for prefix in self.common_prefixes:
+            if normalized_desc.startswith(prefix):
+                normalized_desc = normalized_desc[len(prefix):].strip()
+        
+        # Try to match known merchants
+        for merchant_key, merchant_info in self.merchant_database.items():
```

Ruff (0.11.9)

Loop control variable `merchant_key` not used within loop body

Rename unused `merchant_key` to `_merchant_key`

(B007)